modules/gltf/gltf_document.cpp

```diff
@@ -3748,6 +3748,13 @@ Error GLTFDocument::_serialize_materials(Ref<GLTFState> p_state) {
 			d["alphaMode"] = "BLEND";
 		}
 
+		Dictionary extensions;
+		if (base_material->get_shading_mode() == BaseMaterial3D::SHADING_MODE_UNSHADED) {
+			Dictionary mat_unlit;
```

Comment:
I wonder if this could just be an empty variant instead as it is never checked. Dictionary initializes a few things that I don't think you would get with another variant

Reply:
Some thing like this you mean ?
`extensions["KHR_materials_unlit"] = Variant();`

It didn't work. It yields a json file that looks like this : 
![2023-03-03_20-47](https://user-images.githubusercontent.com/3624853/222813033-71c6c95b-c2f1-450a-aa94-8c03b9dd68c7.png)

But the [spec](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_materials_unlit/README.md) defines it as an empty {object}, not a null : 

![2023-03-03_20-49](https://user-images.githubusercontent.com/3624853/222813242-b04d188c-5773-4966-ba46-c03347c17482.png)

And when importing the gltf file with the variant into blender, it fails, which is expected.

![2023-03-03_20-50](https://user-images.githubusercontent.com/3624853/222813472-4cd21409-c80c-4523-8f6d-63080b112d06.png)

Reply:
Thanks for checking and for referring me to the spec. Indeed it looks like the spec requires an empty dictionary. In hindsight that makes sense as they may make additions to the extension in the future (unlikely for this particular extension, but still).